### ISSUE TITLE
The condition #<CHUNGA:INPUT-CHUNKING-UNEXPECTED-END-OF-FILE 4060090DA0> occurred

### ISSUE THREAD

**Issue by tdrhq** [fukamachi/dexador#121]
Test case:

```
(uiop:slurp-input-stream 'string (dex:get "https://news.lk" :force-binary t :want-stream t))

debugger invoked on a CHUNGA:INPUT-CHUNKING-UNEXPECTED-END-OF-FILE in thread
#<THREAD "main thread" RUNNING {1001878383}>:
  Condition CHUNGA:INPUT-CHUNKING-UNEXPECTED-END-OF-FILE was signalled.

Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.

restarts (invokable by number or by possibly-abbreviated name):
  0: [ABORT] Exit debugger, returning to top level.

((:METHOD CHUNGA::FILL-BUFFER (CHUNGA:CHUNKED-INPUT-STREAM)) #<CHUNGA:CHUNKED-INPUT-STREAM {1003ACCBF3}>) [fast-method]
   source: (ERROR 'INPUT-CHUNKING-UNEXPECTED-END-OF-FILE :STREAM STREAM)

```

That signals the error that I mentioned in the title. Tested on both SBCL and Lispworks 8.0 on Linux.

I think it has something to do with the character encoding, but I'm not 100% sure. Drakma appears to work correctly, which seems to suggest that the website isn't misbehaving:

```
[1] CL-USER> (uiop:slurp-input-stream 'string (drakma:http-request "https://news.lk" :force-binary t :want-stream t))
"<!DOCTYPE html>
<html prefix="og: http://ogp.me/ns#" xmlns="http://www.w3.org/1999/xhtml" xml:lang="en-gb" lang="en-gb" dir="ltr">
    <head>
        <meta http-equiv="X-UA-Compatible" content="IE=edg...[sly-elided string of length 135177]"
```

**Comment by fukamachi** [comment]
Confirmed the issue, and seems like a bug.
FYI, it doesn't occur if `:keep-alive nil` is specified.

**Comment by ajberkley** [comment]
We are handling chunked data wrong.  The keep-alive-chunked stream should push more of its work to chunga.  CRLF is valid data in a chunked stream.  @fukamachi  Should I fix this or are you working on it?

**Comment by ajberkley** [comment]
I will fix this.

**Comment by ajberkley** [comment]
Fixed in PR #124 
You can pull from https://github.com/ajberkley/dexador if you want to try it out.
